### DISCUSSION TITLE
Allow customization of cilium values

### DISCUSSION THREAD

**Discussion by nmehlei** [vitobotta/hetzner-k3s#467] [Ideas]
I've successfully set up multiple clusters with Cilium as CNI, yet the current customization are quite limited. We are using Cilium as an ingress controller, and changed the helm values for Cilium like this, which is working very well:
<img width="585" alt="SCR-20241014-ocip" src="https://github.com/user-attachments/assets/df0d2957-6a28-425a-8fed-db6e37b61fb4">

Unfortunately, any invocation of `hetzner-k3s` to perform cluster changes (e.g. changing node count) resets the helm values for cilium, which also - besides other changes - removes the Load Balancer for Cilium's Ingress Controller and therefore stops all Ingresses from working until the Ingress Controller is activated and the Load Balancer is re-created again.

I would propose that the possible configuration values for Cilium are extended to the ones visible in the screenshot. Some values, like the load balancer annotations, can be automatically set if using Cilium as an Ingress Controller, as the LB would not work otherwise anyway.

For reference, the currently possible values are being defined [here](https://github.com/vitobotta/hetzner-k3s/blob/60b862b3105d6a7362f5754ee83b5f91a2014984/src/kubernetes/software/cilium.cr#L22-L41).

**Comment by vitobotta** [comment]
I understand your use case. I am adding a note to the backlog about it; what I am thinking is to allow a config setting for the path of a yaml file with overrides for the Helm chart. I can't promise when I can do this though since I have several things planned for the next release and am quite busy with the day job at the moment.

**Reply by nmehlei** [reply]
Hey @vitobotta, do you have an update on this by any chance? I completely understand that this is not on the top of the priority list, but I believe this is crucial for production-level cilium usage.

**Reply by vitobotta** [reply]
To be honest, you're the only one who's asked for this, so I've focused on other more urgent tasks. But if adding a configuration option to set the path for Helm value files would work for you, I can include it in the next release.
